pkg/util/cache/cache.go

```diff
@@ -19,14 +19,16 @@ var (
 )
 
 type cacheItem[K comparable, V any] struct {
-	key      K
-	value    V
-	pinCount atomic.Int32
+	key        K
+	value      V
+	pinCount   atomic.Int32
+	needReload bool
 }
 
 type (
 	Loader[K comparable, V any]    func(key K) (V, bool)
 	Finalizer[K comparable, V any] func(key K, value V) error
+	Reloader[K comparable, V any]  func(key K) (V, bool)
```

Comment:
nit: it would be much cleaner if we could reuse the `Loader` concept in reloading. I guess the cache users have convenient ways to differentiate their _reload_ from _load_.

Reply:
> nit: it would be much cleaner if we could reuse the `Loader` concept in reloading. I guess the cache users have convenient ways to differentiate their _reload_ from _load_.

agree~